pkgs/development/python-modules/nbqa/default.nix

```
@@ -0,0 +1,64 @@
+{ autopep8
+, black
+, buildPythonPackage
+, fetchPypi
+, flake8
+, ipython
+, isort
+, jupytext
+, lib
+, mypy
+, pylint
+, pythonAtLeast
+, pythonOlder
+, pyupgrade
+, ruff
+, tokenize-rt
+, tomli
+}:
+buildPythonPackage rec {
+  pname = "nbqa";
+  version = "1.7.0";
+
+  # TODO: Implement support for older Python 3.8 and 3.9, as supported by the project. Requires fixing jupyter-ydoc build.
+  disabled = pythonOlder "3.10" || pythonAtLeast "3.12";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-EXESrW1hj/E6/FukHKD/IcI/CEpt94dfsEt/LAGhNsQ=";
+  };
+
+  passthru.optional-dependencies = {
+    # TODO: Add blacken-docs; it's not yet packaged in nixpkgs.
+    all = [
+      black
+      flake8
+      isort
+      jupytext
+      mypy
+      pylint
+      pyupgrade
+      ruff
+    ];
+    black = [ black ];
+    flake8 = [ flake8 ];
+    isort = [ isort ];
+    jupytext = [ jupytext ];
+    mypy = [ mypy ];
+    pylint = [ pylint ];
+    pyupgrade = [ pyupgrade ];
+    ruff = [ ruff ];
+  };
+
+  propagatedBuildInputs = [ autopep8 ipython tokenize-rt tomli ] ++ passthru.optional-dependencies.all;
```

```suggestion
  propagatedBuildInputs = [ autopep8 ipython tokenize-rt tomli ];
```
If we use this as a package, we should not propagate optional-dependencies.

---

Without the optional dependencies the package is a bit useless, because it's a wrapper for those other packages. How is this normally solved in a user-friendly way? We'd have to somehow communicate to the user that they need to install additional packages to do anything useful.

---

I've removed these, but I'm still not sure what to do about the optional dependencies.

---

Based on our discussion on Matrix I've reintroduced these.